### PR TITLE
feat(docker): setup development infrastructure under docker compose

### PR DESCRIPTION
**Pull request by lelouvincx**
## Description
Working...

Setup docker compose for development team, including:
- [x] Databases
- [x] Backend
- Frontend + Tests will be moved to another PR

## How Has This Been Tested?

Runs well manually under local.

## Related Issue(s)

Issue #35 

## Checklist:

- [x] All of the automated tests pass
- [x] All PR comments are addressed and marked resolved
- [x] If there are migrations, they have been rebased to latest main
- [x] If there are new dependencies, they are added to the requirements
- [x] If there are new environment variables, they are added to all of the deployment methods
- [x] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [x] Docker images build and basic functionalities work
- [x] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: chatbot-core/backend/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.vscode/
+.python-version
+.gitignore
+README.md
+requirements.txt
```
Comment (tuantran0910) [inline chatbot-core/backend/.dockerignore:5]: If using `uv`, consider removing `requirements.txt`. Same for building Docker image (copy `uv.lock` into Docker container and run `uv.sync`).

Comment (tuantran0910) [inline chatbot-core/backend/.dockerignore:5]: If using `uv`, consider removing `requirements.txt`. Same for building Docker image (copy `uv.lock` into Docker container and run `uv.sync`).

File: chatbot-core/backend/Dockerfile [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+WORKDIR /app
+RUN uv sync --frozen --no-cache
 
 # Expose the port on which the application will run
 EXPOSE 5000
 
 # Run the application
-CMD [ "uvicorn", "main:app", "--host", "0.0.0.0", "--port", "5000" ]
\ No newline at end of file
+CMD [ "/app/.venv/bin/fastapi", "run", "app/main.py", "--host", "0.0.0.0", "--port", "5000" ]
```
Comment (tuantran0910) [inline chatbot-core/backend/Dockerfile:21]: Change to command `uv run uvicorn main:app --reload --port 5000`.

File: chatbot-core/backend/app/models/api.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
 from pydantic import BaseModel
 from typing import Optional, Any
 
-from utils.error_handler import ErrorCodesMappingNumber
+from app.utils.error_handler import ErrorCodesMappingNumber
 
 
 class APIResponse(BaseModel):
+    status_code: int = 200
```
Comment (tuantran0910) [inline chatbot-core/backend/app/models/api.py:8]: Don't need this, you can remove it. I will include `status_code` in the parameter of router (e.g. `router.post("/", status_code=status.HTTP_201_SUCCESSFULLY)`

File: chatbot-core/backend/app/settings/__init__.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from .constants import Constants
+from .secrets import Secrets
+
+
+__all__ = [
+    "Constants",
+    "Secrets",
+]
```
Comment (tuantran0910) [inline chatbot-core/backend/app/settings/__init__.py:8]: What is this used for ?

File: chatbot-core/backend/app/utils/api_response.py [hunk 5]
```diff
@@ -13,6 +14,10 @@ def __init__(self) -> None:
 
         self.resp = resp
 
+    def set_status_code(self, status_code: int) -> "BackendAPIResponse":
+        self.resp.status_code = status_code
+        return self
+
```
Comment (tuantran0910) [inline chatbot-core/backend/app/utils/api_response.py:20]: Just remove, don't need it.

File: chatbot-core/backend/app/settings/__init__.py (hunk 4 above)
Comment (lekiet1214) [reply chatbot-core/backend/app/settings/__init__.py:8]: > What is this used for ?
This means the setting module will return two exports `Constant` and `Secrets`.
Only when used with `import *` tho, which is not recommended.

File: chatbot-core/backend/.dockerignore (hunk 1 above)
Comment (lelouvincx) [reply chatbot-core/backend/.dockerignore:5]: Bruh I know, just dockerignore

File: chatbot-core/backend/app/models/api.py (hunk 3 above)
Comment (lelouvincx) [reply chatbot-core/backend/app/models/api.py:8]: Ok.

File: chatbot-core/backend/app/settings/__init__.py (hunk 4 above)
Comment (lelouvincx) [reply chatbot-core/backend/app/settings/__init__.py:8]: As @lekiet1214 said, I saw commonly from LlamaIndex library, so I just copy that.

File: chatbot-core/backend/app/utils/api_response.py (hunk 5 above)
Comment (lelouvincx) [reply chatbot-core/backend/app/utils/api_response.py:20]: Nooooooooooooooooooooo
Remove causes error

Comment (lelouvincx) [reply chatbot-core/backend/app/utils/api_response.py:20]: image

File: chatbot-core/backend/.dockerignore (hunk 1 above)
Comment (tuantran0910) [reply chatbot-core/backend/.dockerignore:5]: Hehe

File: chatbot-core/backend/app/settings/__init__.py (hunk 4 above)
Comment (tuantran0910) [reply chatbot-core/backend/app/settings/__init__.py:8]: Ok good.

File: chatbot-core/backend/app/utils/api_response.py (hunk 5 above)
Comment (tuantran0910) [reply chatbot-core/backend/app/utils/api_response.py:20]: Ok I know this command. Just simply delete method `set_status_code` at each place where the class `BackendAPIResponse` appears.

File: chatbot-core/backend/Dockerfile (hunk 2 above)
Comment (lelouvincx) [reply chatbot-core/backend/Dockerfile]: https://github.com/TinhHoaSolutions-EzHR/chatbot/pull/49/commits/8467b47af2ba86a562c1d2b9ba2c0af763469260

File: chatbot-core/backend/app/utils/api_response.py (hunk 5 above)
Comment (lelouvincx) [reply chatbot-core/backend/app/utils/api_response.py]: Ok https://github.com/TinhHoaSolutions-EzHR/chatbot/pull/49/commits/be1eb40bea1aa10eef0b8fe8a23169b7192ba88c

**Review by tuantran0910 — APPROVED** [review]

**Pull request merged**